### PR TITLE
Fixes #28328 - add DHCP options setting

### PR DESCRIPTION
**Pull request by lzap**
This allows defining them freely.

### CODE REVIEW

**Comment by lzap** [comment]
Sure, other than that any objections?

**Comment by lzap** [comment]
Oh I remember the reason for JSON - it can easily be provided by the installer as a simple string. YAML can't be provided as a oneliner, can it?

**Comment by ekohl** [comment]
I would hope we could use [to_yaml](https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/REFERENCE.md#to_yaml) for that. Asking the user to provide a JSON encoded string is also not a very good experience.

**Comment by lzap** [comment]
Rebased, it's now YAML.

**Comment by lzap** [comment]
Fixed rubocop, how's this?

**Comment by lzap** [comment]
Now for real!

File: .rubocop.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 inherit_from: .rubocop_todo.yml
 
 AllCops:
-  TargetRubyVersion: 2.0
+  TargetRubyVersion: 2.4
```
Comment (mmoll) [inline .rubocop.yml]: I don't think smart-proxy is on 2.5+, yet, or was it already SCLed?

Comment (lzap) [reply .rubocop.yml]: Oh thanks, rebased this was an accident.

**Comment by lzap** [comment]
One more time! Jeeeez.

**Comment by ekohl** [comment]
I'd recommend to split off a PR just to get it to build. I'd also be happy to help you switch it from Travis to Jenkins.

**Comment by lzap** [comment]
Fixed it finally.

**Comment by lzap** [comment]
Any chance to get this through?

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,7 @@
 ---
 rvm:
   - 2.0
-  - 2.1
-  - 2.2
-  - 2.3
-  - 2.4
-  - 2.5
```
Comment (ezr-ondrej) [inline .travis.yml]: the `2.0` and `2.6` is sufficient? Don't we want at least `2.3`? Just asking though.

Comment (ekohl) [reply .travis.yml]: Foreman develop runs on 2.0 (EL7) and 2.5 (Debian Buster & Ubuntu Bionic).

Comment (ezr-ondrej) [reply .travis.yml]: Ah, ok :+1:

**Comment by ares** [comment]
Any more comments? Otherwise I'm merging later today.

File: .travis.yml (hunk 2 above)
Comment (mmoll) [reply .travis.yml]: Please keep 2.5 in there, as Debian and Ubuntu use it.

**Comment by ekohl** [comment]
In the future I'd recommend splitting this because it makes reviewing easier. First a commit that only makes sure the tests pass (possibly as a separate commit), then a commit that adds the new feature. In this case I'd even add a third for the env var that allows disabling SSL verification. https://theforeman.org/2019/03/merge-time-review-complexity.html describes why that increases the chance of a quick merge.

**Comment by lzap** [comment]
Rebased.

**Pull request merged**